cpp/unit/ability.h

```diff
@@ -1,28 +1,53 @@
-// Copyright 2014-2014 the openage authors. See copying.md for legal info.
+// Copyright 2014-2015 the openage authors. See copying.md for legal info.
 
 #ifndef OPENAGE_UNIT_ABILITY_H_
 #define OPENAGE_UNIT_ABILITY_H_
 
 #include <bitset>
 #include <memory>
+#include <type_traits>
+#include <unordered_map>
 
 #include "../coord/phys3.h"
+#include "resource.h"
 
 namespace openage {
```

Comment:
shouldn't this be the `openage::unit` namespace or something? otherwise the `has_hitpoints` etc functions will be in the root `namespace openage`.